packages/next/src/shared/lib/router/utils/is-bot.ts

```diff
@@ -1,6 +1,19 @@
-const BOT_UA_RE =
-  /Googlebot|Mediapartners-Google|AdsBot-Google|googleweblight|Storebot-Google|Google-PageRenderer|Bingbot|BingPreview|Slurp|DuckDuckBot|baiduspider|yandex|sogou|LinkedInBot|bitlybot|tumblr|vkShare|quora link preview|facebookexternalhit|facebookcatalog|Twitterbot|applebot|redditbot|Slackbot|Discordbot|WhatsApp|SkypeUriPreview|ia_archiver/i
+// Bot crawler that will spin up a headless browser
+const HEADLESS_BOT_UA_RE =
+  /Googlebot|Google-PageRenderer|AdsBot-Google|googleweblight|Storebot-Google|Bingbot|BingPreview|applebot|facebookexternalhit|facebookcatalog|Twitterbot|LinkedInBot|Slackbot|Discordbot|WhatsApp|SkypeUriPreview/i
+
+// Static fetcher bot crawler that doesn't spin up a headless browser or execute JS
+const STATIC_FETCHER_BOT_UA_RE =
+  /Mediapartners-Google|Slurp|DuckDuckBot|baiduspider|yandex|sogou|bitlybot|tumblr|vkShare|quora link preview|redditbot|ia_archiver/i
+
+function isHeadlessBrowserBot(userAgent: string) {
+  return HEADLESS_BOT_UA_RE.test(userAgent)
+}
+
+function isNonBrowserBot(userAgent: string) {
+  return STATIC_FETCHER_BOT_UA_RE.test(userAgent)
```

Comment:
let's be consistent with the naming -- ie `STATIC_FETCHER_BOT` should match how we're describing this function `isNonBrowserBot` and not introduce new terms.

Potentially also worth describing these as "HTML-limited bots" to match our internal discussions around it.